src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleAbstractController.php

```diff
@@ -26,37 +26,32 @@
 
 namespace PrestaShopBundle\Controller\Admin\Improve\Modules;
 
-use PrestaShop\PrestaShop\Core\Addon\AddonsCollection;
+use PrestaShop\PrestaShop\Core\Module\ModuleCollection;
+use PrestaShop\PrestaShop\Core\Module\ModuleRepositoryInterface;
 use PrestaShopBundle\Controller\Admin\FrameworkBundleAdminController;
 use PrestaShopBundle\Security\Voter\PageVoter;
 
 abstract class ModuleAbstractController extends FrameworkBundleAdminController
 {
     public const CONTROLLER_NAME = 'ADMINMODULESSF';
 
-    /**
-     * Common method of alerts & updates routes for getting template variables.
-     *
-     * @param string $type Type of alert to display (to_configure / to_update ...)
-     *
-     * @return array
-     */
-    protected function getNotificationPageData($type)
+    public const CONFIGURABLE_MODULE_TYPE = 'to_configure';
+    public const UPDATABLE_MODULE_TYPE = 'to_update';
+    public const TOTAL_MODULE_TYPE = 'count';
+
+    final protected function getNotificationPageData(ModuleCollection $moduleCollection): array
```

Comment:
What do you define it as a `final` ?

Reply:
So it doesn't get overridable by child classes

Reply:
But why this method can't be overriden ?